flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/consumer/RemoteInputChannelTest.java

```diff
@@ -992,16 +993,37 @@ public void testConcurrentRecycleAndRelease2() throws Exception {
 	// ---------------------------------------------------------------------------------------------
 
 	private RemoteInputChannel createRemoteInputChannel(SingleInputGate inputGate)
-			throws IOException, InterruptedException {
+		throws IOException, InterruptedException {
+
+		return createRemoteInputChannel(inputGate, InputChannelTestUtils.StubMemorySegmentProvider.getInstance());
+	}
 
-		return createRemoteInputChannel(inputGate, mock(PartitionRequestClient.class), 0, 0);
+	private RemoteInputChannel createRemoteInputChannel(
+		SingleInputGate inputGate,
+		MemorySegmentProvider memorySegmentProvider)
+		throws IOException, InterruptedException {
+
+		return createRemoteInputChannel(
+				inputGate, mock(PartitionRequestClient.class), 0, 0, memorySegmentProvider);
```

Comment:
better avoid mock `PartitionRequestClient`.

The same issue for creating `RemoteInputChannel` in various units, if we do not need unify the `createPartitionRequestClient` with the constructing of `RemoteInputChannel`, all these `createRemoteInputChannel` could be covered by one `InputChannelBuilder`.